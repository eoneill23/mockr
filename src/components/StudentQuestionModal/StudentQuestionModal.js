import React from 'react';

export const StudentQuestionModal = ({ body, collapseModal, collapsed }) => {
  return (
    <section>
      <button
        onClick={() => collapseModal(!collapsed)}
      >
        Collapse
      </button>
      <p>{body}</p>
    </section>
  )
}

export default StudentQuestionModal;