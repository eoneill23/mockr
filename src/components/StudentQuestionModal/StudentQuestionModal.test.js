import React from 'react';
import { render } from '@testing-library/react';
import StudentQuestionModal from './StudentQuestionModal';
import { MockedProvider } from '@apollo/react-testing';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('StudentQuestionModal', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = render(
      <MockedProvider addTypeName={false}>
        <StudentQuestionModal body={'Great job'} collapseModal={jest.fn()} collapsed={true} />
      </MockedProvider>
    );
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});