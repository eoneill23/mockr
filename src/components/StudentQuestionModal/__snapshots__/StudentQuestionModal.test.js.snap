// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentQuestionModal should match the snapshot with all data passed in correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <button>
          Collapse
        </button>
        <p>
          Great job
        </p>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <button>
        Collapse
      </button>
      <p>
        Great job
      </p>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
